From 105305a486ae9f171def221ad971d607604a4770 Mon Sep 17 00:00:00 2001
From: Jan Friesse <jfriesse@redhat.com>
Date: Thu, 6 Dec 2012 10:47:49 +0100
Subject: [PATCH] On places with POLLERR check also POLLNVAL

POLLNVAL is returned on invalid FD. We must also check it.

Signed-off-by: Jan Friesse <jfriesse@redhat.com>
Reviewed-by: Fabio M. Di Nitto <fdinitto@redhat.com>
---
 exec/coroipcs.c |    2 +-
 lib/coroipcc.c  |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/exec/coroipcs.c b/exec/coroipcs.c
index 1f3fa09..1adb7ae 100644
--- a/exec/coroipcs.c
+++ b/exec/coroipcs.c
@@ -1617,7 +1617,7 @@ int coroipcs_handler_dispatch (
 	/*
 	 * If an error occurs, request exit
 	 */
-	if (revent & (POLLERR|POLLHUP)) {
+	if (revent & (POLLERR|POLLHUP|POLLNVAL)) {
 		ipc_disconnect (conn_info);
 		return (0);
 	}
diff --git a/lib/coroipcc.c b/lib/coroipcc.c
index a1bbbdf..30b1474 100644
--- a/lib/coroipcc.c
+++ b/lib/coroipcc.c
@@ -902,7 +902,7 @@ coroipcc_dispatch_get (
 		error = CS_ERR_TRY_AGAIN;
 		goto error_put;
 	}
-	if (poll_events == 1 && (ufds.revents & (POLLERR|POLLHUP))) {
+	if (poll_events == 1 && (ufds.revents & (POLLERR|POLLHUP|POLLNVAL))) {
 		error = CS_ERR_LIBRARY;
 		goto error_put;
 	}
-- 
1.7.1

